### PR TITLE
Spacy Tokenization Complete

### PR DESCRIPTION
**Pull request by JJModern**
## Describe your changes
Spacy Tokenization Complete

Just finished a module implementing a Tokenizer class that tokenizes text. 
The Tokenizer class provides methods for initializing the required 
components and processing text data. It is designed to handle tokenization
tasks efficiently and effectively in a user-friendly manner.

The below is an example of how the text is tokenized

case_id,speaker,speaker_type,conversation_id,id,text,tokens
1955_71,earl_warren,J,13127,13127__0_000,"number 71, lonnie affronti versus united states of america. mr. murphy.","['number', '71', ',', 'lonnie', 'affronti', 'versus', 'united', 'states', 'of', 'america', '.', 'mr', '.', 'murphy', '.']"
1955_71,harry_f_murphy,A,13127,13127__0_001,"may it please the court. we are here by writ of certiorari to the eighth circuit. there is one question to be decided in this case, decided carefully. upon sentence to consecutive sentences or terms by a district court. the defending pattern started the service of a first sentence. thus, the district court thereafter have jurisdiction to suspend the execution of the remaining sentences and place the defendant on probation.","['may', 'it', 'please', 'the', 'court', '.', 'we', 'are', 'here', 'by', 'writ', 'of', 'certiorari', 'to', 'the', 'eighth', 'circuit', '.', 'there', 'is', 'one', 'question', 'to', 'be', 'decided', 'in', 'this', 'case', ',', 'decided', 'carefully', '.', 'upon', 'sentence', 'to', 'consecutive', 'sentences', 'or', 'terms', 'by', 'a', 'district', 'court', '.', 'the', 'defending', 'pattern', 'started', 'the', 'service', 'of', 'a', 'first', 'sentence', '.', 'thus', ',', 'the', 'district', 'court', 'thereafter', 'have', 'jurisdiction', 'to', 'suspend', 'the', 'execution', 'of', 'the', 'remaining', 'sentences', 'and', 'place', 'the', 'defendant', 'on', 'probation', '.']"
1955_71,william_o_douglas,J,13127,13127__0_002,consecutive sentences.,"['consecutive', 'sentences', '.']"

## Checklist before requesting a review

- [x] The code runs successfully.

```
Here is some console output

isort "supreme_court_predictions"/ test/
Skipped 1 files
black "supreme_court_predictions"/ test/
All done! ✨ 🍰 ✨
15 files left unchanged.

—
-----------------------------------
Your code has been rated at 9.93/10

—

python -m "supreme_court_predictions" --clean-data
oh, what up?
Data will be saved to: 
/supreme-court-ml-predictions/supreme_court_predictions/data/clean_convokit/
Spacy module successfully loaded.
Spacy Tokenization Complete
```
- [x] I ran `make lint` and have made the bulk of changes that it has requested.
- [x] The `Lint` GitHub Action step is passing.
  - Manually run `make format` to correct the errors.

### CODE REVIEW

**Review by FedericoDM — APPROVED** [review]
Beautiful

File: supreme_court_predictions/statistics/tokenizer.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        cwd = os.getcwd()
+        self.local_path = (
+            cwd.replace("\\", "/")
+            + "/supreme_court_predictions/data/clean_convokit/"
+        )
+        print(f"Data will be saved to: \n{self.local_path}")
+
+        try:
+            self.nlp = spacy.load("en_core_web_sm", disable=["parser", "ner"])
+        except OSError:
```
Comment (chanteriam) [inline supreme_court_predictions/statistics/tokenizer.py:36]: Nice!

File: pyproject.toml [hunk 2]
```diff
@@ -19,6 +19,7 @@ isort = "^5.12.0"
 nltk = "^3.8.1"
 openpyxl = "^3.1.2"
 scikit-learn = "^1.2.2"
+spacy = "^3.5.2"
```
Comment (michplunkett) [inline pyproject.toml:22]: Appreciate the add in Poetry!

File: supreme_court_predictions/statistics/tokenizer.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        """
+        Initializes the Tokenizer class by setting up the local path
+        and loading the spaCy model.
+        """
+        # Get local directory
+        cwd = os.getcwd()
+        self.local_path = (
+            cwd.replace("\\", "/")
+            + "/supreme_court_predictions/data/clean_convokit/"
+        )
```
Comment (michplunkett) [inline supreme_court_predictions/statistics/tokenizer.py:31]: I may make this util function today for funsies.

**Pull request merged**